slatedb/src/db.rs

```diff
@@ -395,14 +393,20 @@ impl DbInner {
     }
 
     pub(crate) async fn flush_memtables(&self) -> Result<(), SlateDBError> {
+        let mut froze_memtable = false;
         {
             let last_flushed_wal_id = self.wal_buffer.recent_flushed_wal_id();
             let mut guard = self.state.write();
             if !guard.memtable().is_empty() {
                 guard.freeze_memtable(last_flushed_wal_id)?;
+                froze_memtable = true;
             }
         }
 
+        if froze_memtable {
+            fail_point!(Arc::clone(&self.fp_registry), "flush-memtable-after-freeze");
```

Comment:
🤔 can this be put directly in line 402? It seems like `froze_memtable` is set to true, then we immediately call this fail_point macro...

Reply:
derp! yeah